File: ios/QuickLookViewManager.m
//
//  QuickLookViewManager.m
//  POCApp
//
//  Created by Geoffrey Xue on 7/22/20.
//

#import "QuickLookView-Bridging-Header.h"

@interface RCT_EXTERN_MODULE(QuickLookViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(fileSource, NSNumber)
RCT_REMAP_VIEW_PROPERTY(url, urlString, NSString)
RCT_EXPORT_VIEW_PROPERTY(fileData, NSString)
RCT_EXPORT_VIEW_PROPERTY(fileType, NSString)
RCT_EXPORT_VIEW_PROPERTY(fileID, NSString)
@end

File: ios/QuickLookViewManager.swift
//
//  QuickLookViewManager.swift
//  POCApp
//
//  Created by Geoffrey Xue on 8/14/20.
//

import Foundation


@objc(QuickLookViewManager)
class QuickLookViewManager : RCTViewManager {
  
  var quicklookView: QuickLookView?
  
  override static func requiresMainQueueSetup() -> Bool {
      return true
  }
  
  override func view() -> UIView! {
    quicklookView = QuickLookView()
    return quicklookView
  }
  
  @objc(generatePreview:fileType:fileID:)
  func generatePreview(src: NSString, fileType: NSString, fileID: NSString) {
    DispatchQueue.main.async {
      self.quicklookView?.generatePreview(src: src, fileType: fileType, fileID: fileID)
    }
  }

}
